src/test/runner.ts

```diff
@@ -243,8 +244,24 @@ export class Runner {
 }
 
 function filterOnly(suite: Suite) {
-  const onlySuites = suite.suites.filter(child => filterOnly(child) || child._only);
-  const onlyTests = suite.specs.filter(spec => spec._only);
+  const suiteFilter = (suite: Suite) => suite._only;
+  const specFilter = (spec: Spec) => spec._only;
+  return filterSuite(suite, suiteFilter, specFilter);
+}
+
+function filterByFocusedLine(suite: Suite, focusedTestFileLines: FilePatternFilter[]) {
+  const testFileLineMatches = (specFileName: string, specLine: number) => focusedTestFileLines.some(({re, line}) => {
+    re.lastIndex = 0;
+    return re.test(specFileName) && line === specLine;
```

Comment:
I wonder how this works with null line? For example, `npx playwright test a.spec.ts:80 b.spec.ts` - will this run one test + all tests?

Reply:
The extension probably won't need it but to be feature-complete I added support and a test for that too.